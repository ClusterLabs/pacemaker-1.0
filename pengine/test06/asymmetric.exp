 <transition_graph cluster-delay="10" stonith-timeout="60s" failed-stop-offset="INFINITY" failed-start-offset="1" batch-limit="30" transition_id="0">
   <synapse id="0">
     <action_set>
       <rsc_op id="11" operation="monitor" operation_key="ebe3fb6e-7778-426e-be58-190ab1ff3dd3:0_monitor_20000" on_node="puma3" on_node_uuid="d61fe2cb-98f0-4815-9b7e-2552f5330ad3">
         <primitive id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3:0" long-id="ms_drbd_poolA:ebe3fb6e-7778-426e-be58-190ab1ff3dd3:0" class="ocf" provider="heartbeat" type="drbd"/>
         <attributes CRM_meta_clone="0" CRM_meta_clone_max="2" CRM_meta_clone_node_max="1" CRM_meta_globally_unique="false" CRM_meta_interval="20000" CRM_meta_master_max="1" CRM_meta_master_node_max="1" CRM_meta_name="monitor" CRM_meta_op_target_rc="8" CRM_meta_role="Master" CRM_meta_stateful="true" CRM_meta_timeout="10000" crm_feature_set="3.0.1" drbd_resource="poolA" drbdconf="/etc/drbd.poolA"/>
       </rsc_op>
     </action_set>
     <inputs/>
   </synapse>
   <synapse id="1">
     <action_set>
       <rsc_op id="15" operation="monitor" operation_key="ebe3fb6e-7778-426e-be58-190ab1ff3dd3:1_monitor_19000" on_node="puma1" on_node_uuid="c1b88737-8c7c-4824-81de-843fe7f2e88a">
         <primitive id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3:1" long-id="ms_drbd_poolA:ebe3fb6e-7778-426e-be58-190ab1ff3dd3:1" class="ocf" provider="heartbeat" type="drbd"/>
         <attributes CRM_meta_clone="1" CRM_meta_clone_max="2" CRM_meta_clone_node_max="1" CRM_meta_globally_unique="false" CRM_meta_interval="19000" CRM_meta_master_max="1" CRM_meta_master_node_max="1" CRM_meta_name="monitor" CRM_meta_role="Slave" CRM_meta_stateful="true" CRM_meta_timeout="10000" crm_feature_set="3.0.1" drbd_resource="poolA" drbdconf="/etc/drbd.poolA"/>
       </rsc_op>
     </action_set>
     <inputs/>
   </synapse>
   <synapse id="2">
     <action_set>
       <rsc_op id="4" operation="monitor" operation_key="drbd_target_poolA_monitor_0" on_node="puma3" on_node_uuid="d61fe2cb-98f0-4815-9b7e-2552f5330ad3">
         <primitive id="drbd_target_poolA" long-id="drbd_target_poolA" class="ocf" provider="vpools" type="iscsi_target"/>
         <attributes CRM_meta_op_target_rc="7" CRM_meta_timeout="40000" crm_feature_set="3.0.1" drbd_resource="poolA" drbdconf="/etc/drbd.poolA" ip="192.168.5.1" nexus_iqn="iqn.2008-05.fr.longphone:poolA" target_role="started"/>
       </rsc_op>
     </action_set>
     <inputs/>
   </synapse>
   <synapse id="3">
     <action_set>
       <rsc_op id="6" operation="monitor" operation_key="drbd_target_poolA_monitor_0" on_node="puma1" on_node_uuid="c1b88737-8c7c-4824-81de-843fe7f2e88a">
         <primitive id="drbd_target_poolA" long-id="drbd_target_poolA" class="ocf" provider="vpools" type="iscsi_target"/>
         <attributes CRM_meta_op_target_rc="7" CRM_meta_timeout="40000" crm_feature_set="3.0.1" drbd_resource="poolA" drbdconf="/etc/drbd.poolA" ip="192.168.5.1" nexus_iqn="iqn.2008-05.fr.longphone:poolA" target_role="started"/>
       </rsc_op>
     </action_set>
     <inputs/>
   </synapse>
   <synapse id="4">
     <action_set>
       <pseudo_event id="2" operation="probe_complete" operation_key="probe_complete">
         <attributes crm_feature_set="3.0.1"/>
       </pseudo_event>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="3" operation="probe_complete" operation_key="probe_complete" on_node="puma3" on_node_uuid="d61fe2cb-98f0-4815-9b7e-2552f5330ad3"/>
       </trigger>
       <trigger>
         <rsc_op id="5" operation="probe_complete" operation_key="probe_complete" on_node="puma1" on_node_uuid="c1b88737-8c7c-4824-81de-843fe7f2e88a"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="5" priority="1000000">
     <action_set>
       <rsc_op id="3" operation="probe_complete" operation_key="probe_complete" on_node="puma3" on_node_uuid="d61fe2cb-98f0-4815-9b7e-2552f5330ad3">
         <attributes CRM_meta_op_no_wait="true" crm_feature_set="3.0.1"/>
       </rsc_op>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="4" operation="monitor" operation_key="drbd_target_poolA_monitor_0" on_node="puma3" on_node_uuid="d61fe2cb-98f0-4815-9b7e-2552f5330ad3"/>
       </trigger>
     </inputs>
   </synapse>
   <synapse id="6" priority="1000000">
     <action_set>
       <rsc_op id="5" operation="probe_complete" operation_key="probe_complete" on_node="puma1" on_node_uuid="c1b88737-8c7c-4824-81de-843fe7f2e88a">
         <attributes CRM_meta_op_no_wait="true" crm_feature_set="3.0.1"/>
       </rsc_op>
     </action_set>
     <inputs>
       <trigger>
         <rsc_op id="6" operation="monitor" operation_key="drbd_target_poolA_monitor_0" on_node="puma1" on_node_uuid="c1b88737-8c7c-4824-81de-843fe7f2e88a"/>
       </trigger>
     </inputs>
   </synapse>
 </transition_graph>

