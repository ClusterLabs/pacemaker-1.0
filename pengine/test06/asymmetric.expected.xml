<?xml version="1.0" encoding="UTF-8"?>
<cib crm_feature_set="2.1" admin_epoch="0" epoch="9698" num_updates="3" dc-uuid="c1b88737-8c7c-4824-81de-843fe7f2e88a" have-quorum="1" remote-tls-port="0" validate-with="pacemaker-1.0">
   <configuration>
     <crm_config>
       <cluster_property_set id="cib-bootstrap-options"><nvpair id="nvpair" name="dc-version" value="0.6.3-node: fd8904c9bc67a2396bb5a7908bc7de0540241b1a"/><nvpair id="nvpair" name="symmetric-cluster" value="false"/><nvpair id="nvpair" name="default-resource-stickiness" value="100"/><nvpair id="nvpair" name="stop-orphan-resources" value="true"/><nvpair id="nvpair" name="stop-orphan-actions" value="true"/><nvpair id="nvpair" name="no-quorum-policy" value="ignore"/><nvpair id="nvpair" name="stonith-enabled" value="false"/><nvpair id="nvpair" name="stonith-action" value="reboot"/><nvpair id="nvpair" name="default-resource-failure-stickiness" value="50"/><nvpair id="nvpair" name="is-managed-default" value="true"/><nvpair id="nvpair" name="cluster-delay" value="10"/><nvpair id="nvpair" name="batch-limit" value="30"/><nvpair id="nvpair" name="default-action-timeout" value="40"/><nvpair id="nvpair" name="remove-after-stop" value="false"/><nvpair id="nvpair" name="pe-error-series-max" value="-1"/><nvpair id="nvpair" name="pe-warn-series-max" value="-1"/><nvpair id="nvpair" name="pe-input-series-max" value="-1"/><nvpair id="nvpair" name="startup-fencing" value="false"/><nvpair id="nvpair" name="start-failure-is-fatal" value="false"/><nvpair id="nvpair" name="last-lrm-refresh" value="1211290949"/></cluster_property_set>
     </crm_config>
     <nodes>
       <node id="d61fe2cb-98f0-4815-9b7e-2552f5330ad3" uname="puma3" type="normal">
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="ROLE_storage" value="true"/></instance_attributes>
       </node>
       <node id="c1b88737-8c7c-4824-81de-843fe7f2e88a" uname="puma1" type="normal">
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="ROLE_storage" value="true"/></instance_attributes>
       </node>
     </nodes>
     <resources>
       <master id="ms_drbd_poolA"><meta_attributes id="master_slave-ms_drbd_poolA.meta"/><meta_attributes id="meta_attributes"><nvpair id="nvpair" name="clone-max" value="2"/><nvpair id="nvpair" name="clone-node-max" value="1"/><nvpair id="nvpair" name="master-max" value="1"/><nvpair id="nvpair" name="master-node-max" value="1"/><nvpair id="nvpair" name="notify" value="yes"/><nvpair id="nvpair" name="globally-unique" value="false"/><nvpair id="nvpair" name="target-role" value="started"/></meta_attributes><primitive id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3" class="ocf" provider="heartbeat" type="drbd"><meta_attributes id="primitive-ebe3fb6e-7778-426e-be58-190ab1ff3dd3.meta"/>
           <instance_attributes id="instance_attributes"><nvpair name="drbd_resource" value="poolA" id="nvpair"/><nvpair name="drbdconf" value="/etc/drbd.poolA" id="nvpair"/></instance_attributes>
           <operations>
             <op name="monitor" interval="20s" id="ed238e4a-f523-49f9-889a-d8e81a74ff81" timeout="10s" role="Master"/>
             <op name="monitor" interval="19s" id="op.auto-2" timeout="10s" role="Slave"/>
           </operations>
         </primitive></master>
       <primitive id="vpool_ip_poolA" class="ocf" provider="heartbeat" type="IPaddr2"><meta_attributes id="primitive-vpool_ip_poolA.meta"><nvpair id="nvpair" name="target-role" value="started"/></meta_attributes>
         <instance_attributes id="instance_attributes"><nvpair name="ip" value="192.168.5.1" id="nvpair"/><nvpair name="nic" value="vlan10" id="nvpair"/><nvpair name="cidr_netmask" value="24" id="nvpair"/><nvpair name="arp_count" value="10" id="nvpair"/><nvpair name="target_role" value="started" id="nvpair"/></instance_attributes>
       </primitive>
       <primitive id="drbd_target_poolA" class="ocf" provider="vpools" type="iscsi_target"><meta_attributes id="primitive-drbd_target_poolA.meta"><nvpair id="nvpair" name="target-role" value="started"/></meta_attributes>
         <instance_attributes id="instance_attributes"><nvpair name="ip" value="192.168.5.1" id="nvpair"/><nvpair name="nexus_iqn" value="iqn.2008-05.fr.longphone:poolA" id="nvpair"/><nvpair name="drbd_resource" value="poolA" id="nvpair"/><nvpair name="drbdconf" value="/etc/drbd.poolA" id="nvpair"/><nvpair name="target_role" value="started" id="nvpair"/></instance_attributes>
       </primitive>
     </resources>
     <constraints>
       <rsc_location id="cfc593e5-c42a-404d-9a89-3cf984e0d810" rsc="ms_drbd_poolA">
         <rule id="rule.auto-1" boolean-op="or" score="0">
           <expression attribute="#uname" operation="eq" value="puma3" id="expression"/>
           <expression attribute="#uname" operation="eq" value="puma1" id="expression"/>
         </rule>
       </rsc_location>
       <rsc_location id="de1e56fb-da27-4620-b317-ed76cc663aa3" rsc="ms_drbd_poolA">
         <rule id="rule.auto-2" boolean-op="and" score="-INFINITY">
           <expression attribute="ROLE_storage" operation="ne" value="true" id="expression"/>
         </rule>
       </rsc_location>
       <rsc_order id="e83470b8-5d7d-4507-bc8e-24842af2135a" first="ms_drbd_poolA" then="vpool_ip_poolA" then-action="start" first-action="promote" score="INFINITY"/>
       <rsc_colocation id="rsc_colocation.auto-1" rsc="vpool_ip_poolA" with-rsc="ms_drbd_poolA" with-rsc-role="Master" score="INFINITY"/>
       <rsc_order id="rsc_order.auto-2" first="vpool_ip_poolA" then="drbd_target_poolA" then-action="start" first-action="start" score="INFINITY"/>
       <rsc_colocation id="rsc_colocation.auto-2" rsc="drbd_target_poolA" with-rsc="vpool_ip_poolA" score="INFINITY"/>
     </constraints>
   </configuration>
   <status>
     <node_state id="c1b88737-8c7c-4824-81de-843fe7f2e88a" uname="puma1" ha="active" in_ccm="true" crmd="online" join="member" expected="member" shutdown="0">
       <lrm id="c1b88737-8c7c-4824-81de-843fe7f2e88a">
         <lrm_resources id="lrm_resources">
           <lrm_resource id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3:0" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-0_monitor_0" operation="monitor" transition-key="6:294:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:7;6:294:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="91" crm_feature_set="2.1" rc-code="7" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
           </lrm_resource>
           <lrm_resource id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3:1" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-1_start_0" operation="start" transition-key="7:295:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:0;7:295:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="92" crm_feature_set="2.1" rc-code="0" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-1_post_notify_start_0" operation="notify" transition-key="42:295:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:0;42:295:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="93" crm_feature_set="2.1" rc-code="0" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-1_pre_notify_promote_0" operation="notify" transition-key="55:296:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:0;55:296:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="95" crm_feature_set="2.1" rc-code="0" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-1_post_notify_promote_0" operation="notify" transition-key="56:296:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:0;56:296:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="96" crm_feature_set="2.1" rc-code="0" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
           </lrm_resource>
           <lrm_resource id="vpool_ip_poolA" type="IPaddr2" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="vpool_ip_poolA_monitor_0" operation="monitor" transition-key="6:296:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:7;6:296:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="94" crm_feature_set="2.1" rc-code="7" op-status="0" interval="0" op-digest="69dc64bdd64bd01b54047bcb5ac9c167"/>
           </lrm_resource>
         </lrm_resources>
       </lrm>
       <transient_attributes id="c1b88737-8c7c-4824-81de-843fe7f2e88a">
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="probe_complete" value="true"/></instance_attributes>
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="master-ebe3fb6e-7778-426e-be58-190ab1ff3dd3:1" value="5"/></instance_attributes>
       </transient_attributes>
     </node_state>
     <node_state id="d61fe2cb-98f0-4815-9b7e-2552f5330ad3" uname="puma3" ha="active" in_ccm="true" crmd="online" join="member" expected="member" shutdown="0">
       <lrm id="d61fe2cb-98f0-4815-9b7e-2552f5330ad3">
         <lrm_resources id="lrm_resources">
           <lrm_resource id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3:0" type="drbd" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-0_monitor_0" operation="monitor" transition-key="4:294:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:7;4:294:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="123" crm_feature_set="2.1" rc-code="7" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-0_start_0" operation="start" transition-key="5:295:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:0;5:295:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="124" crm_feature_set="2.1" rc-code="0" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-0_post_notify_start_0" operation="notify" transition-key="40:295:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:0;40:295:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="125" crm_feature_set="2.1" rc-code="0" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-0_pre_notify_promote_0" operation="notify" transition-key="47:296:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:0;47:296:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="127" crm_feature_set="2.1" rc-code="0" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-0_promote_0" operation="promote" transition-key="10:296:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:0;10:296:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="128" crm_feature_set="2.1" rc-code="0" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
             <lrm_rsc_op id="ebe3fb6e-7778-426e-be58-190ab1ff3dd3-0_post_notify_promote_0" operation="notify" transition-key="48:296:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:0;48:296:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="129" crm_feature_set="2.1" rc-code="0" op-status="0" interval="0" op-digest="79f4a52c49cbdbd4614cebcf8a70d315"/>
           </lrm_resource>
           <lrm_resource id="vpool_ip_poolA" type="IPaddr2" class="ocf" provider="heartbeat">
             <lrm_rsc_op id="vpool_ip_poolA_monitor_0" operation="monitor" transition-key="4:296:861f189c-f814-4aa3-8e14-b084dd4d0913" transition-magic="0:7;4:296:861f189c-f814-4aa3-8e14-b084dd4d0913" call-id="126" crm_feature_set="2.1" rc-code="7" op-status="0" interval="0" op-digest="69dc64bdd64bd01b54047bcb5ac9c167"/>
           </lrm_resource>
         </lrm_resources>
       </lrm>
       <transient_attributes id="d61fe2cb-98f0-4815-9b7e-2552f5330ad3">
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="fail-count-6d55db96-eb90-49eb-94e0-aeedef2cff51:0" value="1"/><nvpair id="nvpair" name="probe_complete" value="true"/></instance_attributes>
         <instance_attributes id="instance_attributes"><nvpair id="nvpair" name="master-ebe3fb6e-7778-426e-be58-190ab1ff3dd3:0" value="100"/></instance_attributes>
       </transient_attributes>
     </node_state>
   </status>
 </cib>
